crates/polars-plan/src/plans/optimizer/projection_pushdown/mod.rs

```diff
@@ -482,6 +482,21 @@ impl ProjectionPushDown {
                     };
                 }
 
+                // File builder has a row index, but projected columns
+                // do not include it, so cull.
```

Comment:
Assumption: any filter that refers to the introduced row index does not get pushed into the Scan.